### PR TITLE
Revert "Upgrade t-digest to latest release (#7029)"

### PR DESCRIPTION
**Pull request by mayankshriv**
We found that the latest T-Digest version the following test fails.
- PreAggregatedPercentileTDigestStarTreeV2Test
- The test uses random seed for generating data, so it doesn't always fail.
- Changing the random seed as follows will make the test fail consistently.

In `BaseStarTreeV2Test, modify line that sets the random seed as follows:`

```
  private static final Random RANDOM = new Random(1);
```

We also confirmed that the problem goes away after reverting back to previous t-digest library version.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

**Comment by mayankshriv** [comment]
@tdunning Tagging you on this PR to notify the issue we found with the t-digest library upgrade (details in the PR description), and had to revert back to previous release.

**Comment by tdunning** [comment]
I'll take a look.

**Review by xiangfu0 — APPROVED** [review]

**Comment by tdunning** [comment]
I have done an analysis here. I believe that the root of the issue is that the t-digest takes the compression factor more seriously now than it did before. That means that there will typically be 27 ± 5 centroids in a digest with compression factor 50. In the failure that I analyzed, there were 42 data points across 23 centroids. The issue observed has several contributory factors:

a) the compression value for the t-digest is smaller than typical (50 instead of 100 or 200).
b) the interpretation of compression parameter for 3.3 is more strict resulting in somewhat smaller digest than before.
c) the requirement that values differ by no more than 5 percent is very optimistic when one digest interpolates and the other doesn't. This will happen differently according to just which points get collected together and can easily differ, particularly right on the border between centroids of weight 1 and 2.
d) the quantile chosen interacted with the number of samples to force interpolation to almost as bad a result as possible.

The summary of this is that the t-digest appears to be functioning correctly and this test is a bit too aggressive. The likely best remedy is to increase the compression value used to 100. Increasing default compression for t-digests across the board should be considered as well.

## Detailed analysis
The general situation is as shown in this figure:
Figure showing interpolation

The code necessary to regenerate the figure can be found in [this gist](https://gist.github.com/tdunning/a10d79bc4cb3834cf2ecea96e70cb9f7#file-figure-r).

The dark line with small circles represents the exact empirical CDF as determined by the actual data. With the t-digest, we use the mid-point rule so the CDF at a value exactly equal to a sample is half-way between the values to the left and right of the sample. In this example, t-digest A merged the samples at 1530 and 1940 into a centroid at 1735 and the samples at 2166 and 2168 into a centroid at 2167. This causes interpolation when the value of quantile(0.12) is computed and the resulting value of 923.8 is shown by the left-most circled X. The correct value of the empirical distribution for this quantile is 1530. The interpolated value is far from this correct value because the quantile desired is so close to the transition from one sample to the next.

In contrast, t-digest B starts merging values slightly above this point so sample 1530 is still alone in a centroid. This means that digest B computes the precisely correct value of 1530 (represented by the right-most circled X). The threshold for the test is 5% of 10,000, however, so the test fails because the difference is 606.2.  The fact that these samples have such a large difference is a major contributor here; if this difference were less than 500, the interpolation could not cause this error. It is the combination of the bad luck of the large difference right at the transition to multi-sample centroids that causes this problem.

**Comment by mayankshriv** [comment]
Thanks @tdunning for the detailed analysis.

@Jackie-Jiang could you revisit the test with suggestions from Ted?

**Comment by Jackie-Jiang** [comment]
@tdunning Thanks for the explanation.
It there a formula to calculate the max error rate from the compression factor (or reversed) so that I can modify the test accordingly? Seems our default compression factor 100 can guarantee 5% error rate, and we also want to test non-default compression factor.

**Comment by tdunning** [comment]
Errors in estimation with the t-digest are usually phrased in terms of quantile error (that is, q, not the samples x). Errors in the sample space are notionally unbounded because the sample distribution can be stretched. In addition, t-digest is normally used when estimating tails of a distribution so the relative error $\Delta q$ / \max(q, 1-q)$ is controlled rather than the absolute error. The result is that error near the median can be orders of magnitude larger than near the extremes.

For uniform distribution, errors in sample space are the same as errors in quantile (subject to scaling by the range).

Regarding error scaling with respect to $q$, the figure below shows that for small values of $\delta$ errors scale roughly with $1/\delta^2$ and then transitions to scaling roughly with $1/\sqrt(\delta)$ ([see this article on t-digest](https://www.sciencedirect.com/science/article/pii/S2665963820300403) for more detailed information). The problem with this statement is that it doesn't really respond to your question because it is, again, focused on extreme $q$.

Scaling of error with compression factor delta

Errors near the median for well-behaved distributions like the uniform distribution should be roughly like $2/\delta$ given that there should be about $\delta/2$ centroids. The actual results will be somewhat better than this to the extent that interpolation works. Here is a figure that shows max error of quantile(x) for selected values of x at various combinations of delta and number of samples. Note that these errors are relative to the exact empirical quantiles. Errors in consistency between different t-digests could, conceivably, be double these numbers. Also, if you looked at a high resolution scan of possible values of x, you would likely find bad cases right at the transition between 1 and 2 samples per centroid. 

Errors as a function of delta and number of samples

Basically, $\delta=100$ has *lots* fewer outlier cases and is likely to be what you need. I note that this is the default value for most of your code. This particular test stands out a bit for using $\delta = 50$.

This last figure is generated from the file `median-error.csv` produced by the `com.tdunning.tdigest.quality.CompareKllTest#testMedianErrorVersusScale` test in the main branch of t-digest. The code to produce the [figure can be found in this gist](https://gist.github.com/tdunning/78a4a8ee17b48d35cd7ec496b2f0d817).
